### DISCUSSION TITLE
blocking probe

### DISCUSSION THREAD

**Discussion by dimitrijejankov** [openucx/ucx#6227] [Q&A] [answered]
Hi so I want to use ucx to replace MPI in my research project. One problem I have encountered is that you only have a non-blocking probe "ucp_tag_probe_nb". What would be the best way to replace calls to "MPI_Mprobe"?

**Comment by brminich** [comment, marked as answer]
Hi,
You can replace `MPI_Mprobe` by `ucp_tag_probe_nb()` with remove flag set to 1,  i. e.
`ucp_tag_probe_nb(worker, tag, tag_mask, 1, info);`
This routine returns msg handle which can further be passed to `ucp_tag_msg_recv_nb` to receive the message.
Also to make it blocking you can use a loop, like it is done in OMPI here:
https://github.com/open-mpi/ompi/blob/master/ompi/mca/pml/ucx/pml_ucx.c#L1029

**Reply by dimitrijejankov** [reply]
Thanks a lot :)
